corehq/apps/geospatial/routing_solvers/mapbox_optimize.py

```diff
@@ -0,0 +1,129 @@
+import json
+import jsonschema
+import requests
+from django.conf import settings
+
+
+def validate_routing_request(request_json):
+    schema = {
+        "type": "object",
+        "properties": {
+            "users": {
+                "type": "array",
+                "items": {
+                    "$ref": "#/definitions/location"
+                }
+            },
+            "cases": {
+                "type": "array",
+                "items": {
+                    "$ref": "#/definitions/location"
+                }
+            }
+        },
+        "definitions": {
+            "location": {
+                "type": "object",
+                "properties": {
+                    "lon": {
+                        "type": ["string", "number"],
+                        "description": "longitude"
+                    },
+                    "lat": {
+                        "type": ["string", "number"],
+                        "description": "latitude"
+                    },
+                    "id": {
+                        "type": "string",
+                        "description": "id or pk"
+                    }
+                },
+                "required": ["lon", "lat", "id"]
+            }
+        },
+        "required": ["users", "cases"]
+    }
+    jsonschema.validate(request_json, schema)
+
+
+def submit_routing_request(request_json):
+    # Submits a request to Mapbox Optimize V2 endpoint
+    # Docs https://docs.mapbox.com/api/navigation/optimization/#submit-a-routing-problem
+    validate_routing_request(request_json)
+    payload = generate_request_payload(request_json)
+    # Todo; consider using a separte ACCESS_TOKEN for optimize API
+    response = requests.post(
+        "https://api.mapbox.com/optimized-trips/v2/"
```

Comment:
Nit: can the API URL string be a constant?

Reply:
It would be useful if this were used in more than one place, but it's used only here. It would add unnecessary imports etc.